src/fugw/mappings/dense.py

```diff
@@ -149,17 +150,53 @@ def fit(
             Ft_val = _make_tensor(target_features_val.T, device=device)
             F_val = torch.cdist(Fs_val, Ft_val, p=2) ** 2
 
+        elif source_features_val is not None and target_features_val is None:
+            raise ValueError(
+                "Source features validation data provided but not target"
+                " features validation data."
+            )
+
+        elif source_features_val is None and target_features_val is not None:
+            raise ValueError(
+                "Target features validation data provided but not source"
+                " features validation data."
+            )
+
         else:
             F_val = None
 
+            # Raise warning if validation feature maps are not provided
+            warnings.warn(
+                "Validation data for feature maps is not provided."
+                " Using training data instead."
+            )
```

Comment:
I don't think we should raise a warning (it will be printed in a majority of cases, although there is no real problem).
Instead, we could log this message if `verbose=True`. You can import fugw's rich console with `from fugw.utils import console` and use it this way:
```suggestion
            console.log(
                "Validation data for feature maps is not provided."
                " Using training data instead."
            )
```